xbmc/settings/Settings.cpp

```diff
@@ -456,6 +456,9 @@ bool CSettings::InitializeDefinitions()
 #elif defined(TARGET_ANDROID)
   if (CFile::Exists(SETTINGS_XML_FOLDER "android.xml") && !Initialize(SETTINGS_XML_FOLDER "android.xml"))
     CLog::Log(LOGFATAL, "Unable to load android-specific settings definitions");
+#elif defined(HAS_LIBAMCODEC)
```

Comment:
> i can add aml-linux.xml and aml-android.xml if its wanted...

Actually, HAS_LIBAMCODEC is always defined on Android, so that doesn't make much sense. It only tell that the codec has been compiled, not that it runs on aml.

You should probably test "aml-present()" instead (https://github.com/OpenELEC/xbmc/blob/aml-fixes/xbmc/utils/AMLUtils.cpp#L36) and use it + "aml-android" / "aml-linux" under TARGET_ANDROID / TARGET_LINUX, respectively.